src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,157 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.NoSuchElementException;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        private Node(Node<T> prev, T element, Node<T> next) {
+            this.value = element;
+            this.prev = prev;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (isEmpty()) {
+            last = new Node<>(null, value, null);
+            first = last;
+        } else {
+            Node<T> newNode = new Node<>(first, value, null);
+            first.next = newNode;
+            first = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (index == 0) {
+            Node<T> newNode = new Node<>(null, value, last);
+            last = newNode;
+            last.next.prev = newNode;
+        } else {
+            Node<T> nodeOfIndex = getNode(index);
+            Node<T> newNode = new Node<>(nodeOfIndex.prev, value, nodeOfIndex);
+            nodeOfIndex.prev.next = newNode;
+            nodeOfIndex.prev = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        if (checkingIndex(index)) {
+            T old = getNode(index).value;
+            getNode(index).value = value;
+            return old;
+        }
+        throw new IndexOutOfBoundsException();
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (index >= 0 && index < size) {
```

Comment:
it would be more logical to create check index method then remove a condition and an exception into there